crates/consensus/debug-client/src/providers/etherscan.rs

```diff
@@ -32,13 +32,17 @@ impl EtherscanBlockProvider {
     /// `BlockNumberOrTag::Earliest`, `BlockNumberOrTag::Pending`, `BlockNumberOrTag::Number(u64)`
     /// are supported.
     pub async fn load_block(&self, block_number_or_tag: BlockNumberOrTag) -> eyre::Result<Block> {
+        let tag = match block_number_or_tag {
+            BlockNumberOrTag::Number(num) => format!("{num:#02x}"),
+            tag => tag.to_string(),
+        };
```

Comment:
Let's open an issue in upstream to see if we should change the `fmt::Display` implementation for this.

Reply:
https://github.com/alloy-rs/alloy/issues/1852